Snowflake.Data/Core/SnowflakeTomlConnectionBuilder.cs

```diff
@@ -0,0 +1,128 @@
+            if (connectionToml != null)
+            {
+                connectionString = GetConnectionStringFromTomlTable(connectionToml);
+            }
+            return connectionString;
+        }
+
+        private string GetConnectionStringFromTomlTable(TomlTable connectionToml)
+        {
+            var connectionStringBuilder = new StringBuilder();
+            var tokenFilePathValue = string.Empty;
+            var isOauth = connectionToml.TryGetValue("authenticator", out var authenticator) && authenticator.ToString().Equals("oauth");
+            foreach (var property in connectionToml.Keys)
+            {
+                if (isOauth && property.Equals("token_file_path", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    tokenFilePathValue = (string)connectionToml[property];
+                    continue;
+                }
+                var mappedProperty = _tomlToNetPropertiesMapper.TryGetValue(property, out var mapped) ? mapped : property;
+                connectionStringBuilder.Append($"{mappedProperty}={(string)connectionToml[property]};");
+            }
+
+            if (!isOauth || connectionToml.ContainsKey("token"))
+                return connectionStringBuilder.ToString();
+
+            var token = LoadTokenFromFile(tokenFilePathValue);
+            if (!string.IsNullOrEmpty(token))
+            {
+                connectionStringBuilder.Append($"token={token};");
+            }
+            else
+            {
+                _logger.Warn("The token has empty value");
+            }
+
+
+            return connectionStringBuilder.ToString();
+        }
+
+        private string LoadTokenFromFile(string tokenFilePathValue)
+        {
+            var tokenFile = !string.IsNullOrEmpty(tokenFilePathValue) && _fileOperations.Exists(tokenFilePathValue) ? tokenFilePathValue : DefaultTokenPath;
+            _logger.Debug($"Read token from file path: {tokenFile}");
+            return _fileOperations.Exists(tokenFile) ? _fileOperations.ReadAllText(tokenFile) : null;
+        }
+
+        private TomlTable GetTomlTableFromConfig(string tomlPath, string connectionName)
+        {
+            if (!_fileOperations.Exists(tomlPath))
+            {
+                return null;
+            }
+
+            var tomlContent = _fileOperations.ReadAllText(tomlPath) ?? string.Empty;
+            var toml = Toml.ToModel(tomlContent);
+            if (string.IsNullOrEmpty(connectionName))
+            {
+                connectionName = _environmentOperations.GetEnvironmentVariable(EnvironmentVariables.SnowflakeDefaultConnectionName, DefaultConnectionName);
+            }
+
+            var connectionExists = toml.TryGetValue(connectionName, out var connection);
+            if (!connectionExists && connectionName != DefaultConnectionName)
+            {
+                throw new Exception("Specified connection name does not exist in connections.toml");
+            }
+
```

Comment:
The implementation is to try to use the toml configuration file if the ConnectionString is not defined. So it tries to use the toml file and the defined connection name; if the user has the default values and is not defined, the connection is not his purpose to use the mechanism, and the normal exception is thrown by the driver when a connection is not defined should be thrown instead of this specific scenario. @sfc-gh-knozderko

Reply:
Removed overload and handled each case separately.

Reply:
maybe just add a comment that for default connection name the user maybe is not interested in toml config feature, but the exception would be thrown anyway in other place, because connection string cannot be empty

Reply:
Added comment.

Reply:
but I would rather like to have comments about WHY not WHAT is done in the code. You wrote about WHAT is happening. Maybe we could rephrase it to `we don't validate the case of connection not found for the default connection name because for such a case it is likely that the user didn't want to use toml configuration, but just forgot to provide the connection string in the connection. The error would be thrown later anyway, because empty string is not a valid connection string` ?